cosmos/operators/_asynchronous/bigquery.py

```diff
@@ -137,3 +141,48 @@ def execute(self, context: Context, **kwargs: Any) -> None:
             super().execute(context=context)
         else:
             self.build_and_run_cmd(context=context, run_as_async=True, async_context=self.async_context)
+        self._store_compiled_sql(context=context)
+
+    @provide_session
+    def _store_compiled_sql(self, context: Context, session: Session = NEW_SESSION) -> None:
+        from airflow.models.renderedtifields import RenderedTaskInstanceFields
+        from airflow.models.taskinstance import TaskInstance
+
+        if not enable_setup_async_task:
+            self.log.info("SQL cannot be made available, skipping registration of compiled_sql template field")
+            return
+        sql = self.get_remote_sql().strip()
+        self.log.info("Executed SQL is: %s", sql)
```

Comment:
Shall we make it debug log, wdyt?

Reply:
I preferred it to be info since it was nice to see the formatted query in the logs, but okay, changed to debug :)